### DISCUSSION TITLE
new feature for mousedown

### DISCUSSION THREAD

**Discussion by JohnXG** [ZeeCoder/use-resize-observer#93] [Ideas]
Wonder if it's worth adding another return value to indicate mousedown status, so the caller can do some CSS stuff based on it for instance.
Anyway, thanks for sharing the library.

**Comment by ZeeCoder** [comment]
Sorry, but that's a very clear no for me.
This library is intentionally lean and composable, and only serves as a very lightweight proxy to ResizeObserver, nothing more.

It's very easy to compose it with a custom hook though, if you want add the extra property for yourself.
See [this codesandbox as an example](https://codesandbox.io/s/use-resize-observer-with-mousedown-mouseup-gclgxj).

I would encourage you to use the hooks separately though, instead of using a merged hook with too many responsibilities.
